packages/core/src/implementations/business/QuestionnaireService.ts

```diff
@@ -201,4 +196,14 @@ export class QuestionnaireService implements IQuestionnaireService {
     // consent contracts that have staked against the questionnaire.
     throw new Error("Method not implemented.");
   }
+
+  private uniqueCids(cids: IpfsCID[]): IpfsCID[] {
+    const uniqueCidsMap = new Map<IpfsCID, boolean>();
```

Comment:
Instead of a Map, you can use a Set here to get the unique CIDs. It's more efficient